bugbug/repository.py

```diff
@@ -411,131 +352,99 @@ def download_commits(repo_dir, date_from):
+        experiences_by_commit["total"][experience_type][commit.node] = len(all_commits)
+        experiences_by_commit["90_days"][experience_type][commit.node] = len(
+            all_commits - before_90_days_commits
+        )
 
-        for directory in directories:
-            directories_touched_prev[commit.node] += directories_touched[directory]
+    for commit in tqdm(commits):
+        days = (commit.pushdate - first_pushdate).days
+        assert days >= 0
 
-            directories_touched[directory] += 1
+        if days not in experiences:
+            assert days not in complex_experiences
+            experiences[days] = copy.deepcopy(experiences[days - 1])
+            complex_experiences[days] = copy.deepcopy(complex_experiences[days - 1])
 
+        update_experiences("author", days, [commit.author])
+        update_experiences("reviewer", days, commit.reviewers)
+
+        # When a file is moved/copied, copy original experience values to the copied path.
         if len(commit.file_copies) > 0:
             for orig, copied in commit.file_copies.items():
-                if orig in path_to_component and copied in path_to_component:
-                    components_touched[path_to_component[copied]] = components_touched[
-                        path_to_component[orig]
-                    ]
-
-                files_touched[copied] = files_touched[orig]
-
                 orig_directories = get_directories(orig)
-
                 copied_directories = get_directories(copied)
 
-                if len(orig_directories) == len(copied_directories):
-                    for i in range(len(orig_directories)):
-                        if orig_directories[i] != copied_directories[i]:
-                            directories_touched[
-                                copied_directories[i]
-                            ] = directories_touched[orig_directories[i]]
-                elif orig_directories and copied_directories:
-                    directories_touched[copied_directories[0]] = directories_touched[
-                        orig_directories[0]
-                    ]
-
-        for i, prev_commit in enumerate(prev_commits_90_days):
-            if (commit.date - prev_commit.date).days <= 90:
-                break
-
-            cut = i
-
-        if cut is not None:
-            prev_commits_90_days = prev_commits_90_days[cut + 1 :]
-
-        components_touched_90_days = defaultdict(int)
-        files_touched_90_days = defaultdict(int)
-        directories_touched_90_days = defaultdict(int)
-        for prev_commit in prev_commits_90_days:
-            components_prev = set(
+                if orig in path_to_component and copied in path_to_component:
+                    orig_component = path_to_component[orig]
+                    copied_component = path_to_component[copied]
+                else:
+                    orig_component = copied_component = None
+
+                for prev_day in complex_experiences.keys():
```

Comment:
nit: move the "for" in the previous "if" to avoid the "else" and check for None in the for loop

Reply:
I'd need to do two for loops this way